src/natcap/invest/urban_nature_access.py

```diff
@@ -347,328 +504,647 @@ def execute(args):
+    elif args['search_radius_mode'] == RADIUS_OPT_POP_GROUP:
+        pop_group_table = utils.read_csv_to_dataframe(
+            args['population_group_radii_table'])
+        search_radii = set(pop_group_table['search_radius_m'].unique())
+        # Build a dict of {pop_group: search_radius_m}
+        search_radii_by_pop_group = dict(
+            pop_group_table[['pop_group', 'search_radius_m']].itertuples(
+                index=False, name=None))
+    else:
+        valid_options = ', '.join(
+            ARGS_SPEC['args']['search_radius_mode']['options'].keys())
+        raise ValueError(
+            "Invalid search radius mode provided: "
+            f"{args['search_radius_mode']}; must be one of {valid_options}")
+
     for search_radius_m in search_radii:
         search_radius_in_pixels = abs(
             search_radius_m / squared_lulc_pixel_size[0])
-        kernel_paths[search_radius_m] = os.path.join(
+        kernel_path = os.path.join(
             intermediate_dir, f'kernel_{search_radius_m}{suffix}.tif')
+        kernel_paths[search_radius_m] = kernel_path
         kernel_tasks[search_radius_m] = graph.add_task(
-            # All kernel creation types have the same function signature
-            kernel_creation_functions[decay_function],
-            args=(search_radius_in_pixels, kernel_paths[search_radius_m]),
-            kwargs={'normalize': False},  # Model math calls for un-normalized
+            _create_kernel_raster,
+            kwargs={
+                'kernel_function': kernel_creation_functions[decay_function],
+                'expected_distance': search_radius_in_pixels,
+                'kernel_filepath': kernel_path,
+                'normalize': False},  # Model math calls for un-normalized
             task_name=(
                 f'Create {decay_function} kernel - {search_radius_m}m'),
-            target_path_list=[kernel_paths[search_radius_m]]
+            target_path_list=[kernel_path]
         )
 
-        # Convolving the population within a non-normalized kernel gives us the
-        # number of people (possibly weighted, depending on the kernel) within
-        # the target search radius.
-        convolved_population_paths[search_radius_m] = os.path.join(
+    # Search radius mode 1: the same search radius applies to everything
+    if args['search_radius_mode'] == RADIUS_OPT_UNIFORM:
+        search_radius_m = list(search_radii)[0]
+        LOGGER.info("Running model with search radius mode "
+                    f"{RADIUS_OPT_UNIFORM}, radius {search_radius_m}")
+
+        decayed_population_path = os.path.join(
             intermediate_dir,
-            f'convolved_population_{search_radius_m}{suffix}.tif')
-        convolved_population_tasks[search_radius_m] = graph.add_task(
-            _convolve_and_set_lower_bounds_for_population,
+            f'decayed_population_within_{search_radius_m}{suffix}.tif')
+        decayed_population_task = graph.add_task(
```

Comment:
Yeah, I thought "decayed" was more consistent with the terminology in the user's guide.  Functionally what the model needs is a distance-weighted sum of pixel values within a given search radius, it just so happens that we're using a non-normalized convolution to do it.  So the convolution feels more like a technical detail rather than something the user cares about.  Decay, on the other hand, is a thing that the user cares about.